diagnostics/diagnostic.go

```diff
@@ -468,6 +474,59 @@ func (d *DiagnosticClient) runSnapshotFilesListListener() {
 	}()
 }
 
+func appendWithCap(list *list.List, cap int, items []time.Duration) {
```

Comment:
suggestion: we could move this to the same file as `BlockMetrics` struct and add receiver funcs that can be used like this (a bit more organised and self-contained):

```
if len(info.Header) > 0 {
    d.blockMetrics.AppendHeaderTimeMetric(info.HeaderTimes)
}
...
d.blockMetrics.AppendBodiesTimeMetric(...)
...
d.blockMetrics.AppendBodiesTimeMetric(...)
...
d.blockMetrics.AppendExecutionStartTimeMetric(...)
...

func (bm *BlockMetrics) AppendHeaderTimeMetrics(times []time.Duration) {
    appendTimeMetricsWithCap(bm.Header, times)
}
```